### ISSUE TITLE
GPS needs App to be open (display unlocked)

### ISSUE THREAD

**Issue by RedAnarchy** [cgeo/cgeo#10452]
## Bug description 

c:geo needs to be on display to track position. The cache-approaching alarm is just working, when the displayis on. After turning off the display, c:geo looses the gps connection and is not tracking/recording the position and also doesn't alarm near the caches. In the apps request permissions, the gps is just allowed to work when the app is active. Maybe this should be set (optional) to the access background location request permission.

### Steps to reproduce the problem

Opening live map, then turning off the display while walking towards a cache. 

### Actual result after these steps 

The alarm should beep, but nothing happenes. The gpx track has gaps and jumps to the position where you unlock the display.

### Expected result after these steps 

Alarm should beep, gpx track should be continuous

## c:geo version 

2021.04.24

## Reproducible 

Yes

## System information 

```
--- System information ---
c:geo version: 2021.04.24

Device:
-------
- Device type: SM-G960F (starltexx, samsung)
- Available processors: 8
- Android version: 10
- Android build: QP1A.190711.020.G960FXXUFFUC6
- Sailfish OS detected: false
- Google Play services: enabled - 21.12.13 (120408-367530751)
- HW acceleration: enabled (default state)

Sensor and location:
-------
- Low power mode: inactive
- Compass capabilities: yes
- Rotation vector sensor: present
- Orientation sensor: present
- Magnetometer & Accelerometer sensor: present
- Direction sensor used: rotation vector

Program settings:
-------
- Hide caches: -
- Hide waypoints: -
- Set language: de_DE (system default)
- System date format: dd.MM.yy
- Debug mode active: no
- Live map mode: true
- OSM multi-threading: true / threads: 4
- Global filter: display all caches
- Last backup: 26. Apr. 2020, 11:18
- Routing mode: Walk
- Map: Kombiniert (Offline)
  - Id: cgeo.geocaching.maps.mapsforge.MapsforgeMapProvider$OfflineMultiMapSource
  - Atts: alps.map: / Map data (c) OpenStreetMap contributors /  / austria.map: / Map data (c) OpenStreetMap contributors /  / baden-wuerttemberg.map: / Map data (c) OpenStreetMap contributors /  / bayern.map: / Map data (c) OpenStreetMap contributors /  / hessen.map: / Map data (c) OpenStreetMap contributors /  / rheinland-pfalz.map: / Map data (c) OpenStreetMap contributors
  - Theme: none

Services:
-------
- Geocaching sites enabled:
   geocaching.com: Logged in (Anmeldung OK) / PREMIUM
   extremcaching.com: Not logged in (Anmeldung…)
   opencaching.de: Logged in (Anmeldung OK)
- Geocaching.com date format: dd/MM/yyyy
- BRouter installed: true / connection available: true
- Installed c:geo plugins: contacts

Permissions & paths:
-------
- Fine location permission: granted
- Write external storage permission: granted
- System internal c:geo dir: /data/user/0/cgeo.geocaching (4,9 GB free) v2 internal isDir(6 entries)
- Legacy User storage c:geo dir: /storage/emulated/0/cgeo (4,9 GB free) v2 external non-removable isDir(5 entries)
- Geocache data: /storage/emulated/0/Android/data/cgeo.geocaching/files/GeocacheData (4,9 GB free) v2 external non-removable isDir(1568 entries)
- Internal theme sync (is turned off): /data/user/0/cgeo.geocaching/MapThemeData (4,9 GB free) v2 internal isDir(0 entries)
- Public Folders: #9
  - BASE: /cgeo (User-Defined)[/cgeo[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo, Av:true, files:>=62, dirs:>=6, totalFileSize:>=45,2 MB, free space: 4,9 GB, files on device: 1740800)
  - OFFLINE_MAPS: /Locus/maps (User-Defined)[/Locus/maps[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3ALocus%2Fmaps::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3ALocus%2Fmaps/document/primary%3ALocus%2Fmaps, Av:true, files:12, dirs:1, totalFileSize:3,2 GB, free space: 4,9 GB, files on device: 1740800)
  - OFFLINE_MAP_THEMES: /Locus/maps/_themes (Default)[/Locus/maps/_themes[PERSISTABLE_FOLDER(OFFLINE_MAPS)#1:p-content://com.android.externalstorage.documents/tree/primary%3ALocus%2Fmaps::/_themes]] (Uri: content://com.android.externalstorage.documents/tree/primary%3ALocus%2Fmaps/document/primary%3ALocus%2Fmaps%2F_themes, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 4,9 GB, files on device: 1740800)
  - LOGFILES: /cgeo/logfiles (Default)[/cgeo/logfiles[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/logfiles]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Flogfiles, Av:true, files:1, dirs:0, totalFileSize:944,3 KB, free space: 4,9 GB, files on device: 1740800)
  - GPX: /cgeo/gpx (Default)[/cgeo/gpx[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/gpx]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Fgpx, Av:true, files:59, dirs:0, totalFileSize:31,0 MB, free space: 4,9 GB, files on device: 1740800)
  - BACKUP: /cgeo/backup (Default)[/cgeo/backup[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/backup]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Fbackup, Av:true, files:2, dirs:1, totalFileSize:13,4 MB, free space: 4,9 GB, files on device: 1740800)
  - FIELD_NOTES: /cgeo/field-notes (Default)[/cgeo/field-notes[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/field-notes]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Ffield-notes, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 4,9 GB, files on device: 1740800)
  - SPOILER_IMAGES: /cgeo/GeocachePhotos (Default)[/cgeo/GeocachePhotos[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/GeocachePhotos]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2FGeocachePhotos, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 4,9 GB, files on device: 1740800)
  - TEST_FOLDER: [Legacy]/data/user/0/cgeo.geocaching/files/unittest (Default)[/data/user/0/cgeo.geocaching/files/unittest[FILE#1:p-file:///data/user/0/cgeo.geocaching/files::/unittest]] (Uri: file:///data/user/0/cgeo.geocaching/files/unittest, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 4,9 GB, files on device: -1)
- Map render theme path: 
- PersistedDocumentUris: #1
- TRACK: null
- Persisted Uri Permissions: #2
  - content://com.android.externalstorage.documents/tree/primary%3ALocus%2Fmaps (21. März, 10:41):RW
  - content://com.android.externalstorage.documents/tree/primary%3Acgeo (21. März, 10:40):RW
- Database: /data/user/0/cgeo.geocaching/databases/data (v94, Size:63,3 MB) on system internal storage
 -Settings: v5, Count:158
--- End of system information ---
```

**Comment by Lineflyer** [comment]
Well, yes:
This would require to request background location permission, which is AFAIK a dedicated permission since Android 11.
According to my knowledge there are strict rules applied to apps requesting such permission. Therefore I am not sure, whether we can/should implement it.

**Comment by moving-bits** [comment]
Not only that, we also need to implement a background service for it, see #7057

**Comment by jakedot** [comment]
I guess this issue can be closed in favor of #7057 then? it's pretty much exactly the same.

**Comment by Lineflyer** [comment]
Yes, thanks for looking it up. No need for duplicates then.

Duplicate of #7057
